packages/components/src/placeholder/index.js

```diff
@@ -15,10 +16,18 @@ import Icon from '../icon';
  * @return {Object}       The rendered placeholder.
  */
 function Placeholder( { icon, children, label, instructions, className, notices, preview, isColumnLayout, ...additionalProps } ) {
-	const classes = classnames( 'components-placeholder', className );
+	const [ resizeListener, { width } ] = useResizeAware();
+	const classes = classnames(
+		'components-placeholder',
+		( width >= 320 ? 'is-large' : '' ),
+		( width >= 160 && width < 320 ? 'is-medium' : '' ),
+		( width < 160 ? 'is-small' : '' ),
```

Comment:
One thing I note here is that in the documentation of `react-resize-aware`, it mentions how `width` would be `null` the first time the component is rendered:

>This object contains the `width` and `height` properties, these properties are going to be `null` before the component rendered, and will return a number after the component rendered.

https://www.npmjs.com/package/react-resize-aware#api

The "fun" thing about `null` is that, in the context of these sorts of comparisons, it's treated the same as zero ([related](https://javascript.info/comparison#strange-result-null-vs-0), [specification](http://www.ecma-international.org/ecma-262/6.0/#sec-tonumber)).

So:

```js
let width = null;
console.log( width < 160 );
// true
```

So we'll always apply the `is-small` class on the first render, even if it ends up being that the element will actually occupy a large width.

I guess my question is:

- _Must_ we have at least one of `is-large`, `is-medium`, or `is-small`, and therefore it would be okay to use `is-small` as the default until an accurate value can be determined?
- Otherwise, would it be more correct to wait to assign any of these modifier classes until the true width can be determined?

Reply:
For context, I think this may be contributing to some intermittent failures in the end-to-end tests, where we're [trying to click "Try again"](https://github.com/WordPress/gutenberg/blob/c079120ce39044e53ea41a6a1a246aeec8195433/packages/e2e-tests/specs/editor/various/embedding.test.js#L241) when an embed fails, but the placement of the button might be shifting around because it is [rendered after the "instructions"](https://github.com/WordPress/gutenberg/blob/c079120ce39044e53ea41a6a1a246aeec8195433/packages/block-library/src/embed/embed-placeholder.js#L37-L44) which are [hidden while `is-small` class is applied](https://github.com/WordPress/gutenberg/blob/c079120ce39044e53ea41a6a1a246aeec8195433/packages/components/src/placeholder/style.scss#L115-L119).

Reply:
I've opened #19825 for continued discussion.